### ISSUE TITLE
False positive reportOverlappingOverload for constrained TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5511]
This was reported in the mypy bug database, but pyright exhibits the same problem.

https://github.com/python/mypy/issues/15683

Here's a minimal repro:

```python
# pyright: strict
from typing import Sequence, overload, AnyStr
AllStr = bytes | str

@overload
def func1(choices: Sequence[AnyStr]) -> AnyStr: ...

@overload
def func1(choices: Sequence[AllStr]) -> AllStr: ...

def func1(choices: Sequence[AllStr]) -> AllStr: ...

def func2(
    strings: tuple[str], byte_strings: tuple[bytes], mixed_strings: tuple[AllStr]
):
    reveal_type(func1(strings))  # str
    reveal_type(func1(byte_strings))  # bytes
    reveal_type(func1(mixed_strings))  # bytes | str, despite the error
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.318, which I just published. It will also be included in a future release of pylance.
